drivers/i2c/i2c_ll_stm32.c

```diff
@@ -173,10 +174,11 @@ static int i2c_stm32_transfer(const struct device *dev, struct i2c_msg *msg,
 	/* Prevent driver from being suspended by PM until I2C transaction is complete */
 #ifdef CONFIG_PM_DEVICE_RUNTIME
 	(void)pm_device_runtime_get(dev);
-#else
-	pm_device_busy_set(dev);
 #endif
 
+	/* Prevent the clocks to be stopped during the i2c transaction */
+	pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
```

Comment:
@gautierg-st Should we add `PM_STATE_SUSPEND_TO_RAM` here ?

Reply:
We'll have to at some point if we want to support Suspend to RAM for I2C, but that alone is not sufficient. We'd also need to reinitialize the I2C registers after exiting S2RAM. So I'd say it's out of scope of this PR and can be done later in a dedicated S2RAM I2C PR.